src/pyodide/BUILD.bazel

```diff
@@ -50,236 +50,4 @@ cc_capnp_library(
-        "url",
-        "vm",
-        "ws",
-        "node:child_process",
-        "node:crypto",
-        "node:fs",
-        "node:path",
-        "node:url",
-        "node:vm",
-    ],
-    format = "esm",
-    output = "generated/emscriptenSetup.js",
-    target = "esnext",
-    deps = ["pyodide.asm.js@rule_js"],
-)
-
-INTERNAL_MODULES = glob(
-    [
-        "internal/*.ts",
-        "internal/topLevelEntropy/*.ts",
-        # The pool directory is only needed by typescript, it shouldn't be used at runtime.
-        "internal/pool/*.ts",
-        "types/*.ts",
-        "types/*/*.ts",
-    ],
-    allow_empty = True,
-)
-
-MODULES = ["python-entrypoint-helper.ts"]
-
-INTERNAL_DATA_MODULES = glob([
-    "internal/*.py",
-    "internal/patches/*.py",
-    "internal/topLevelEntropy/*.py",
-]) + [
-    "generated/python_stdlib.zip",
-    "generated/pyodide.asm.wasm",
-    "generated/emscriptenSetup.js",
-]
-
-wd_ts_bundle(
-    name = "pyodide",
-    eslintrc_json = "eslint.config.mjs",
-    import_name = "pyodide",
-    internal_data_modules = INTERNAL_DATA_MODULES,
-    internal_modules = INTERNAL_MODULES,
-    js_deps = [
-        "generated/emscriptenSetup",
-        "pyodide.asm.wasm@copy",
-        "python_stdlib.zip@copy",
-    ],
-    lint = False,
-    modules = MODULES,
-    schema_id = "0xbcc8f57c63814005",
-    tsconfig_json = "tsconfig.json",
-)
-
-genrule(
-    name = "pyodide.capnp.bin@rule",
-    srcs = [
-        ":pyodide.capnp",
-        "//src/workerd/jsg:modules.capnp",
-    ] + [
-        ":pyodide-" + m.replace("/", "_").removesuffix(".ts")
-        for m in INTERNAL_MODULES
-        if not m.endswith(".d.ts")
-    ] + [
-        ":pyodide-" + m.replace("/", "_").removesuffix(".ts") + ".d.ts"
-        for m in INTERNAL_MODULES
-        if not m.endswith(".d.ts") and m.endswith(".ts")
-    ] + [
-        ":pyodide_" + m.replace("/", "_").removesuffix(".ts")
-        for m in MODULES
-    ] + [
-        ":pyodide_" + m.replace("/", "_").removesuffix(".ts") + ".d.ts"
-        for m in MODULES
-    ] + [
-        ":pyodide-" + m.replace("/", "_")
-        for m in INTERNAL_DATA_MODULES
-        if m.endswith(".py")
-    ] + [
-        ":pyodide-internal_generated_emscriptenSetup.js",
-        ":pyodide-internal_generated_pyodide.asm.wasm",
-        ":pyodide-internal_generated_python_stdlib.zip",
-    ],
-    outs = ["pyodide.capnp.bin"],
-    cmd = " ".join([
-        # Annoying logic to deal with different paths in workerd vs downstream.
-        # Either need "-I src" in workerd or -I external/workerd/src downstream
-        "INCLUDE=$$(stat src > /dev/null 2>&1 && echo src || echo external/workerd/src);",
-        "$(execpath @capnp-cpp//src/capnp:capnp_tool)",
-        "eval",
-        "$(location :pyodide.capnp)",
-        "pyodideBundle",
-        "-I $$INCLUDE",
-        "-o binary",
-        "> $@",
-    ]),
-    tools = ["@capnp-cpp//src/capnp:capnp_tool"],
-    visibility = ["//visibility:public"],
-)
+main()
```

Comment:
Maybe if we could separate everything that is in that `main` function into multiple functions, but otherwise I think it's better if it's just inside this file.

Reply:
See https://github.com/cloudflare/workerd/pull/3807 where it becomes useful that this is a function.